src/stream/src/executor/sink.rs

```diff
@@ -526,9 +577,8 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
                 match msg? {
                     Message::Watermark(w) => yield Message::Watermark(w),
                     Message::Chunk(mut chunk) => {
-                        // Compact the chunk to eliminate any useless intermediate result (e.g. UPDATE
-                        // V->V).
-                        if compact_chunk {
+                        // Compact the chunk to eliminate any unnecessary updates to external systems.
+                        if !skip_compact {
```

Comment:
Previous `compact_chunk` is the same as `!skip_compact`, which is also `!sink.is_blackhole()`